File: 2020/day1/day1.py
with open('day1', 'r') as fd: # read file
    out = fd.readlines()      # str
out = [int(x) for x in out]   # str -> int

# Part 1 -- two numbers add up to 2020
for x in out:
    for y in out:
        if x + y == 2020:
            print(x*y)
# Part 2 -- three numbers add up to 2020
for x in out:
    for y in out:
        for z in out:
            if x + y + z == 2020:
                print(x*y*z)

File: 2021/day1.go
package main

import (
    "bufio"
    "fmt"
    "io"
    "strconv"
    "os"
)

// Read day1 input
func ReadInts(r io.Reader) ([]int, error) {
    // read file descriptor
    scanner := bufio.NewScanner(r)
    scanner.Split(bufio.ScanWords)

    var result []int
    for scanner.Scan() {
        // str2int
        x, err := strconv.Atoi(scanner.Text())
        if err != nil { return result, err }

        // add to array
        result = append(result, x)
    }
    return result, scanner.Err()
}

func inc(ints []int) (int) {
    var count int = 0
    for i := 0; i < len(ints)-1; i++ {
        if ints[i] < ints[i+1] {
            count++
        }
    }
    return count
}

func main() {
    fd, _ := os.Open("day1")
    ints, _ := ReadInts(fd)

    // Part 1
    fmt.Println(inc(ints))
    
    var win_sum []int
    var sum = 0
    // Part 2
    const win = 3;
    for i := 0; i < len(ints)-win+1; i++ {

        sum = 0
        for j := 0; j < win; j++ {
            sum += ints[i+j]
        }
        win_sum = append(win_sum, sum)
    }
    fmt.Println(inc(win_sum))
}
